### ISSUE TITLE
ContentResource still uses the var REST_API_CONTENT_ALLOW_FRONT_END_SAVING

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#17510]
When we removed the old legacy variables from the ContentResource, we missed one here. It is preventing front end content saving, unless the variable `REST_API_CONTENT_ALLOW_FRONT_END_SAVING` is set to true

https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/rest/ContentResource.java#L1562

We need to allow front end content saving if the front end user has permissions (if they are anonymous then also `CONTENT_APIS_ALLOW_ANONYMOUS=WRITE` needs to be set).

**Comment by wezell** [comment]
It looks like when we changed something in the contentAPI that is breaking checkins by CMS_ANON. See the test above.

**Comment by wezell** [comment]
Here is the error in the logs (once you get rid of `REST_API_CONTENT_ALLOW_FRONT_END_SAVING`) 
```
17:33:05.593  ERROR business.ESContentletAPIImpl - User didn't have permissions to the object the category was being assigned to or to the category with inode f6d809b9-a45b-4661-8c77-13408f9b6fd1
com.dotmarketing.exception.DotSecurityException: User didn't have permissions to the object the category was being assigned to or to the category with inode f6d809b9-a45b-4661-8c77-13408f9b6fd1
	at com.dotmarketing.portlets.categories.business.CategoryAPIImpl.setParents_aroundBody60(CategoryAPIImpl.java:559) ~[classes/:?]
	at com.dotmarketing.portlets.categories.business.CategoryAPIImpl$AjcClosure61.run(CategoryAPIImpl.java:1) ~[classes/:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[classes/:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[classes/:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[classes/:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[classes/:?]
	at com.dotmarketing.portlets.categories.business.CategoryAPIImpl.setParents(CategoryAPIImpl.java:528) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.moveContentCategories(ESContentletAPIImpl.java:4770) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.moveContentDependencies(ESContentletAPIImpl.java:4599) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.internalCheckin(ESContentletAPIImpl.java:3973) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.lambda$36(ESContentletAPIImpl.java:3439) ~[classes/:?]
	at com.dotcms.concurrent.lock.StripedLockImpl.tryLock(StripedLockImpl.java:100) ~[classes/:?]
	at com.dotcms.concurrent.lock.StripedLockImpl.tryLock(StripedLockImpl.java:55) ~[classes/:?]
	at com.dotcms.concurrent.lock.IdentifierStripedLock.tryLock(IdentifierStripedLock.java:16) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.checkin_aroundBody136(ESContentletAPIImpl.java:3438) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl$AjcClosure137.run(ESContentletAPIImpl.java:1) ~[classes/:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[classes/:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[classes/:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[classes/:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.checkin(ESContentletAPIImpl.java:3400) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.checkin(ESContentletAPIImpl.java:7329) ~[classes/:?]
	at com.dotcms.content.elasticsearch.business.ESContentletAPIImpl.checkin(ESContentletAPIImpl.java:7345) ~[classes/:?]
	at com.dotmarketing.portlets.contentlet.business.ContentletAPIInterceptor.checkin(ContentletAPIInterceptor.java:176) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet.executeAction_aroundBody0(SaveContentActionlet.java:70) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet$AjcClosure1.run(SaveContentActionlet.java:1) ~[classes/:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[classes/:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[classes/:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[classes/:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.actionlet.SaveContentActionlet.executeAction(SaveContentActionlet.java:54) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin_aroundBody164(WorkflowAPIImpl.java:2333) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl$AjcClosure165.run(WorkflowAPIImpl.java:1) ~[classes/:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[classes/:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) ~[classes/:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) ~[classes/:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireWorkflowPostCheckin(WorkflowAPIImpl.java:2313) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.lambda$78(WorkflowAPIImpl.java:3038) ~[classes/:?]
	at com.dotcms.util.ThreadContextUtil.wrapVoidNoReindex(ThreadContextUtil.java:85) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireContentWorkflow_aroundBody178(WorkflowAPIImpl.java:3038) ~[classes/:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl$AjcClosure179.run(WorkflowAPIImpl.java:1) ~[classes/:?]
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149) ~[aspectjrt-1.8.10.jar:?]
	at com.dotcms.aspects.aspectj.AspectJDelegateMethodInvocation.proceed(AspectJDelegateMethodInvocation.java:42) ~[classes/:?]
	at com.dotmarketing.db.LocalTransaction.wrapReturnWithListeners(LocalTransaction.java:59) [classes/:?]
	at com.dotcms.aspects.interceptors.WrapInTransactionMethodInterceptor.invoke(WrapInTransactionMethodInterceptor.java:23) [classes/:?]
	at com.dotcms.aspects.aspectj.WrapInTransactionAspect.invoke(WrapInTransactionAspect.java:41) [classes/:?]
	at com.dotmarketing.portlets.workflows.business.WorkflowAPIImpl.fireContentWorkflow(WorkflowAPIImpl.java:3005) [classes/:?]
	at com.dotcms.rest.ContentResource.saveContent(ContentResource.java:1592) [classes/:?]
	at com.dotcms.rest.ContentResource.singlePUTandPOST(ContentResource.java:1555) [classes/:?]
	at com.dotcms.rest.ContentResource.singlePOST(ContentResource.java:1512) [classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_202]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_202]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_202]
```

**Comment by wezell** [comment]
PR: #17526

**Comment by fabrizzio-dotCMS** [comment]
Reviewing for internal QA

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.1 // Postgres // FF
